src/commcare_cloud/ansible/plugins/inventory/csv.py

```diff
@@ -0,0 +1,195 @@
+TYPE_LIST = 'L'
+
+
+class InventoryModule(BaseInventoryPlugin):
+    """Host inventory parser for ansible using csv files."""
+
+    NAME = 'csv'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        self._hosts = set()
+
+    def verify_file(self, path):
+        """Verify if file is usable by this plugin, base does minimal accesability check"""
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            file_name, ext = os.path.splitext(path)
+            if not ext or ext in ('.csv', '.CSV'):
+                valid = True
+        return valid
+
+    def parse(self, inventory, loader, path, cache=False):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        try:
+            if self.loader:
+                (b_data, private) = self.loader._get_file_contents(path)
+            else:
+                b_path = to_bytes(path, errors='surrogate_or_strict')
+                with open(b_path, 'rb') as fh:
+                    b_data = fh.read()
+
+            # Faster to do to_text once on a long string than many
+            # times on smaller strings
+            data = to_text(b_data, errors='surrogate_or_strict').splitlines()
+
+            self._parse(data)
+        except Exception as e:
+            raise AnsibleParserError(e)
+
+    def _parse(self, lines):
+        row_groups = self._parse_row_groups(lines)
+        for row_group in row_groups:
+            rows = list(csv.DictReader(row_group))
+            if 'hostname' in rows[0]:
+                self._parse_hosts(rows)
+            elif 'group' in rows[0]:
+                self._parse_groups(rows)
+
+    def _parse_hosts(self, rows):
+        for row in rows:
+            groups = self._get_host_groups(row)
+            host_vars = self._get_host_vars(row)
+            host = row['host_address']
+            host_group = row['hostname']
+            self.inventory.add_group(host_group)
+            self.inventory.add_host(host, host_group)
+            for group in groups:
+                self.inventory.add_group(group)
+                self.inventory.add_child(group, host_group)
+
+            self.populate_host_vars([host], host_vars)
+
+    def _get_host_groups(self, row):
+        return [
+            val.strip() for key, val in row.items()
+            if key.startswith('group') and val.strip()
+        ]
+
+    def _get_host_vars(self, row):
+        vars = {'hostname': row['hostname']}
+        for key, val in row.items():
+            if 'var' in key and val:
+                item_type, name = key.split('.') if '.' in key else ('S', key)
+                name = name.split(' ')[1]
+                vars[name] = conv_str2value(item_type, val)
+        return vars
+
+    def _parse_groups(self, rows):
+        for row in rows:
+            group = row['group']
+            self.inventory.add_group(group)
+            var_name, type_, raw_val = row['var'], row['type'], row['val']
```

Comment:
nit: you use `item_type` above to refer to the same thing (which i generally like better than the trailing underscore). consistency would be nice.